x-pack/plugins/aiops/public/application/url_state/log_rate_analysis.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { WindowParameters } from '@kbn/aiops-utils';
+
+import { getDefaultAiOpsListState, type AiOpsFullIndexBasedAppState } from './common';
+
+export interface LogRateAnalysisPageUrlState {
+  pageKey: 'logRateAnalysis';
+  pageUrlState: LogRateAnalysisAppState;
+}
+/**
+ * To avoid long urls, we store the window parameters in the url state not with
+ * their full parameters names but with abbrevations. `windowParameters2AppState` and
+ * `appState2WindowParameters` are used to transform the data structure.
+ */
+export interface LogRateAnalysisAppState extends AiOpsFullIndexBasedAppState {
+  /** Window parameters */
+  wp?: {
+    /** Baseline minimum value */
+    bMin: number;
+    /** Baseline maximum value */
+    bMax: number;
+    /** Deviation minimum value */
+    dMin: number;
+    /** Deviation maximum value */
+    dMax: number;
+  };
+}
+
+/**
+ * Transforms a full window parameters object to the abbreviated url state version.
+ */
+export const windowParameters2AppState = (wp?: WindowParameters): LogRateAnalysisAppState['wp'] =>
+  wp && {
+    bMin: wp.baselineMin,
+    bMax: wp.baselineMax,
+    dMin: wp.deviationMin,
+    dMax: wp.deviationMax,
+  };
+
+/**
+ * Transforms an abbreviated url state version of window parameters to its full version.
+ */
+export const appState2WindowParameters = (
```

Comment:
Nit: I think we should keep function names clear and consistent - imo it would be preferable to have a slightly longer function name `appStateToWindowParameters` than to introduce a number that has no meaning but to take the place of a word.

Reply:
Good suggestion, updated in [907d63b](https://github.com/elastic/kibana/pull/171398/commits/907d63b568677c47d361b23fc7b13c2ed0a06707).